addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -260,10 +260,11 @@ def _get_l10n_in_edi_partner_details(self, partner, set_vat=True, set_phone_and_
             if is_overseas is true then pin is 999999 and GSTIN(vat) is URP and Stcd is .
             if pos_state_id is passed then we use set POS
         """
+        zip_digits = self._l10n_in_edi_extract_digits(partner.zip)
         partner_details = {
             "Addr1": partner.street or "",
             "Loc": partner.city or "",
-            "Pin": int(self._l10n_in_edi_extract_digits(partner.zip)),
+            "Pin": zip_digits and int(zip_digits) or "",
```

Comment:
Avoiding a direct traceback is good, but shouldn't it still raise an error? What would the government say if the file contain empty zipcode?

Reply:
it's required for India and for out side of Indian (export) it's always `999999` and we set it here https://github.com/odoo/odoo/blob/15.0/addons/l10n_in_edi/models/account_edi_format.py#L289

Reply:
Exactly, if it's required for India then it should raise an error or warning when you try to create a document with an indian partner that has no zipcode right?

Reply:
already done here https://github.com/odoo/odoo/blob/15.0/addons/l10n_in_edi/models/account_edi_format.py#L232

Reply:
OK, sorry I missed that part :p